aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -65,6 +88,14 @@ protected AnalyticsClient getAnalyticsClient() {
         return analyticsClient;
     }
 
+    /**
+     * Accessor method for pinpoint targeting client.
+     * @return returns pinpoint targeting client.
+     */
+    protected TargetingClient getTargetingClient() {
```

Comment:
This class is `final`, and can't be extended. So `protected` isn't of much value, here. I think maybe you want just default package-local access, here?